### ISSUE TITLE
Flatpak version

### ISSUE THREAD

**Issue by razzeee** [ExpressLRS/ExpressLRS-Configurator#390]
Hey everybody, 

I had a go at a flatpak version and would like to document this here.

You can find the code here:
https://github.com/razzeee/com.github.expresslrs.ExpressLRSConfigurator

It seems to startup fine and building firmware seems to work.

Here's the problems I'm aware of:

1. Can't select a device to flash to, as `udevadm` doesn't seem available. This might be possible with `flatpak-spawn --host` as far as I found out, but that would need to be patched in the app I guess.

2. There are some mesa/nvidia/zinc warnings at the start of the app, not sure what they affect

So basically creating firmware seems to work, but no flashing for now.

**Comment by razzeee** [comment]
Both git and python are there and work fine as far as I can tell.

Electron builder might be complicated, as I ended up building this from the `.deb` and not from source unfortunately.

When trying to build from source, specifically when installing (probably post install) one of the used dependencies tries to download / call out to some website. But flatpak builds must be done offline for security reasons, so that fails.

I'll try to have another look and see if the builder approach might change things.

**Comment by razzeee** [comment]
I don't think using electron-builder is the correct way forward, after a quick look.

Regarding the device selection, it seems to be caused by the serialport lib and there's already an issue over there for it, but hasn't gained much traction.

https://github.com/serialport/node-serialport/issues/2306

**Comment by razzeee** [comment]
https://github.com/flathub/flathub/pull/3655

**Comment by razzeee** [comment]
The flatpak repo is here now: https://github.com/flathub/org.expresslrs.ExpressLRSConfigurator

And there should be a build available in a few hours (~4h)

**Comment by razzeee** [comment]
Here we go https://flathub.org/apps/details/org.expresslrs.ExpressLRSConfigurator

https://beta.flathub.org/apps/details/org.expresslrs.ExpressLRSConfigurator
